cg/services/illumina_services/illumina_post_processing_service/illumina_post_processing_service.py

```diff
@@ -26,37 +29,51 @@ def __init__(self, status_db: Store, housekeeper_api: HousekeeperAPI, dry_run: b
         self.dry_run: bool = False
 
     @staticmethod
-    def store_illumina_flow_cell(
-        flow_cell: FlowCellDirectoryData,
-        store: Store,
+    def get_illumina_flow_cell(
+        flow_cell_dir_data: FlowCellDirectoryData,
     ) -> IlluminaFlowCell:
         """
-        Create flow cell from the parsed and validated flow cell data.
+        Create Illumina flow cell from the parsed and validated flow cell directory data.
         And add the samples on the flow cell to the model.
         """
-        model: str | None = flow_cell.run_parameters.get_flow_cell_model()
+        model: str | None = flow_cell_dir_data.run_parameters.get_flow_cell_model()
         new_flow_cell = IlluminaFlowCell(
-            internal_id=flow_cell.id, type=DeviceType.ILLUMINA, model=model
+            internal_id=flow_cell_dir_data.id, type=DeviceType.ILLUMINA, model=model
         )
-        return store.add_illumina_flow_cell(new_flow_cell)
+        return new_flow_cell
 
     @staticmethod
-    def store_illumina_sequencing_metrics(flow_cell: IlluminaFlowCell) -> None:
+    def get_illumina_sequencing_run(
+        flow_cell_dir_data: FlowCellDirectoryData,
+    ) -> IlluminaSequencingRun:
         """Store illumina run metrics in the status database."""
-        pass
+        metrics_service = IlluminaMetricsService()
+        return metrics_service.create_illumina_sequencing_run(flow_cell_dir_data)
 
-    @staticmethod
-    def store_illumina_sample_sequencing_metrics():
+    def get_illumina_sample_sequencing_metrics(
+        self,
+        flow_cell_dir_data: FlowCellDirectoryData,
+    ) -> list[IlluminaSampleSequencingMetrics]:
         """Store illumina sample sequencing metrics in the status database."""
-        pass
+        metrics_service = IlluminaMetricsService()
+        return metrics_service.create_sample_sequencing_metrics_for_flow_cell(
+            flow_cell_directory=flow_cell_dir_data.path, store=self.status_db
+        )
 
-    def store_illumina_flow_cell_data(self, flow_cell: FlowCellDirectoryData) -> None:
+    def store_illumina_flow_cell_data(self, flow_cell_dir_data: FlowCellDirectoryData) -> None:
         """Store flow cell data in the status database."""
-        flow_cell: IlluminaFlowCell = self.store_illumina_flow_cell(
-            flow_cell=flow_cell, store=self.status_db
+        flow_cell: IlluminaFlowCell = self.get_illumina_flow_cell(
+            flow_cell_dir_data=flow_cell_dir_data
+        )
+        sequencing_run: IlluminaSequencingRun = self.get_illumina_sequencing_run(flow_cell_dir_data)
+        sample_metrics: list[IlluminaSampleSequencingMetrics] = (
+            self.get_illumina_sample_sequencing_metrics(flow_cell_dir_data)
         )
-        self.store_illumina_sequencing_metrics(flow_cell)
-        self.store_illumina_sample_sequencing_metrics()
+        sequencing_run.device = flow_cell
+        sequencing_run.sample_metrics = sample_metrics
```

Comment:
can make this into a "relate_illumina_models" if required. but we have to split getting the models filled, relating them and commitng in this way I think. Otherwise passing models around will become very messy,